src/neo/IO/Caching/DataCache.cs

```diff
@@ -165,7 +165,7 @@ public IEnumerable<(TKey Key, TValue Value)> Find(byte[] key_prefix = null)
                     ))
                     .OrderBy(p => p.KeyBytes, ByteArrayComparer.Default)
                     .ToArray();
-                cachedKeySet = dictionary.Select(p => p.Key).ToHashSet();
+                cachedKeySet = new HashSet<TKey>(cached.Select(p => p.Item2));
```

Comment:
This will lost `TrackState.Deleted` items?

Reply:
Maybe we can use `new HashSet<TKey>(cached.Select(p => p.Item2).concat(changeSet))`.

Reply:
Fixed.